tests/test_tensor/test_model.py

```diff
@@ -249,23 +307,95 @@ def run_1d_row_tp():
         if i > 5:
             break
 
+def run_bert_1d():
+    get_components_func = non_distributed_component_funcs.get_callable('bert')
+    model_builder, train_dataloader, _, optimizer_class, criterion = get_components_func()
+    device = get_current_device()
+    
+    set_seed(1)
+    with ColoInitContext(device=device):
+        model = model_builder(checkpoint=True)
+    
+    # parallel_action_list_row = [
+    #     ParallelAction(priority=1, compute_pattern=ComputePattern.TP1DRow_Linear, parallel_mode=ParallelMode.PARALLEL_1D)
+    # ]
+    # spec_row = TensorSpec(parallel_action_list_row)
+
+    parallel_action_list_col = [
+        ParallelAction(priority=1, compute_pattern=ComputePattern.TP1DCol_Linear, parallel_mode=ParallelMode.PARALLEL_1D)
+    ]
+    spec_col = TensorSpec(parallel_action_list_col)
+
+    parallel_action_list_embedding_col = [
+        ParallelAction(priority=1, compute_pattern=ComputePattern.TP1DCol_Embedding, parallel_mode=ParallelMode.PARALLEL_1D)
+    ]
+    spec_embedding_col = TensorSpec(parallel_action_list_embedding_col)
+
+    for name, p in model.named_parameters():
+        print(f"{name}: {p}")
+
+    for name, p in named_params_with_colotensor(model):
+        if not isinstance(p, ColoTensor):
+            continue
+        #print(name)
+        if 'classifier' in name and ('weight' in name or 'bias' in name):
+            p.set_spec(spec_col)
```

Comment:
Just a side note regarding the current design regarding customized operations, will such usage be an over-kill? I have two questions as listed below:
1. does the user need to know the name of the parameters for replacement?
2. this seems like one-size-fits-all and does not cater to the demand of customization? For example, I only want to keep some linear layers parallelized and the others not, is it possible to achieve this?

Reply:
1. You are right. Current usage is unsuitable to be exposed to users. It's just a verification that we have a method to provide parellel without changing models at all. Maybe specs should be inserted by analyzing op graph automatically in the final.
2. Yes, every op is splitted if we handle output carefully(just like insert a reshape op into the ops with different shape).

Reply:
Ok, understand